astropy/table/table.py

```diff
@@ -335,6 +362,88 @@ def __get__(self, instance, owner_cls):
         return info
 
 
+class HiddenColumns(MetaAttribute):
+    """Maintain tuple of table column names that are hidden from print output.
+
+    This is a descriptor that inherits from MetaAttribute so that the attribute
+    value is stored in the table meta['__attributes__'].
+    """
+    def __get__(self, instance, owner_cls):
+        # For getting from class not an instance
+        if instance is None:
+            return self
+
+        hidden_columns = instance.__dict__.get('hidden_columns')
+        if hidden_columns is None:
+            hidden_columns = deepcopy(self)
+            instance.__dict__['hidden_columns'] = hidden_columns
+        # We set _parent_ref on every call, since if one makes copies of
+        # instances, 'hidden_columns' will be copied as well, which will lose the
+        # reference.
+        hidden_columns._instance_ref = weakref.ref(instance)
+        return hidden_columns
+
+    def __set__(self, instance, names):
+        if isinstance(names, str):
+            names = [names]
+        if names is None:
+            # Remove hidden_columns from the meta['__attributes__'] dict.
+            # Subsequent access will just return None.
+            del instance.hidden_columns
+        else:
+            return super().__set__(instance, tuple(names))
+
+    def __call__(self):
+        instance = self._instance_ref()
+        return super().__get__(instance, instance.__class__)
+
+    def __repr__(self):
+        if hasattr(self, '_instance_ref'):
+            out = f'<{self.__class__.__name__} name={self.name} value={self()}>'
+        else:
+            out = super().__repr__()
+        return out
+
+    def _add_remove_setup(self, names):
+        if isinstance(names, str):
+            names = [names]
+        instance = self._instance_ref()
+        value = super().__get__(instance, instance.__class__)
+        if value is None:
+            value = ()
+        return instance, names, value
+
+    def add(self, names):
```

Comment:
With `add` and `remove`, one gets part of the behaviour of a set, but not all.  An option might be to actually store a `set` inside and then do something like:
```
def __getattr__(self, attr):
    value = self()
    if hasattr(value, attr):
        return getattr(value, attr)
```
Then even things like `table.hidden_columns |= {'a'}` would work...
(Or possibly one can be a mixin of `MetaAttribute` and `set`??)

Note: obviously, *not* important - since that would just make `.add` and `.remove` work the same way as they are now.

Reply:
The external API and implementation are more like `tuple` now instead of `set`. The `add` and `remove` method names are following the general schema of `Table` methods, not `set`.

One benefit of tuple, which may be only a benefit to my own version of tidiness, is that the YAML repr of a set is ugly (`!!set {a:null, b:null, c:null}`).

But I agree there is something compelling to the idea of `set` underneath, so I'll think about it some more.